gitclone/gitclone.go

```diff
@@ -215,45 +276,16 @@ func Execute(cfg Config) error {
 		}
 	}
 
-	checkoutArg := getCheckoutArg(cfg.Commit, cfg.Tag, cfg.Branch)
-	if checkoutArg != "" {
-		log.Infof("\nExporting git logs\n")
-
-		for format, env := range map[string]string{
-			`%H`:  "GIT_CLONE_COMMIT_HASH",
-			`%s`:  "GIT_CLONE_COMMIT_MESSAGE_SUBJECT",
-			`%b`:  "GIT_CLONE_COMMIT_MESSAGE_BODY",
-			`%an`: "GIT_CLONE_COMMIT_AUTHOR_NAME",
-			`%ae`: "GIT_CLONE_COMMIT_AUTHOR_EMAIL",
-			`%cn`: "GIT_CLONE_COMMIT_COMMITER_NAME",
-			`%ce`: "GIT_CLONE_COMMIT_COMMITER_EMAIL",
-		} {
-			if err := printLogAndExportEnv(gitCmd, format, env, maxEnvLength); err != nil {
-				return newStepError(
-					"export_envs_failed",
-					fmt.Errorf("gitCmd log failed: %v", err),
-					"Exporting envs failed",
-				)
-			}
-		}
-
-		count, err := runner.RunForOutput(gitCmd.RevList("HEAD", "--count"))
-		if err != nil {
-			return newStepError(
-				"count_commits_failed",
-				fmt.Errorf("get rev-list failed: %v", err),
-				"Counting commits failed",
-			)
-		}
-
-		log.Printf("=> %s\n   value: %s\n", "GIT_CLONE_COMMIT_COUNT", count)
-		if err := tools.ExportEnvironmentWithEnvman("GIT_CLONE_COMMIT_COUNT", count); err != nil {
-			return newStepError(
-				"export_envs_commit_count_failed",
-				fmt.Errorf("envman export failed: %v", err),
-				"Exporting commit count env failed",
-			)
+	if ref := checkoutStrategy.getBuildTriggerRef(); ref != "" {
+		fmt.Println()
+		log.Infof("Exporting commit details")
+		if err := exportCommitInfo(gitCmd, ref, isPR, maxEnvLength); err != nil {
+			return newStepError("export_envs_failed", err, "Exporting envs failed")
 		}
+	} else {
+		fmt.Println()
+		log.Warnf(`Can not export commit information like commit message and author as it is not available.
+This may happen when using Bitbucket with the "Manual merge" Input set to 'yes' (using a Diff file).`)
```

Comment:
Nit: `Input` -> `input` doesn't need capitalization if I'm not mistaken.

Reply:
Indded, fixed.